### PR TITLE
Fix issue in teardown when folder vsphere doesn't exist

### PR DESCRIPTION
**Pull request by petr-balogh**
Fixes: #9669

### CODE REVIEW

**Comment by petr-balogh** [comment]
Verification destroy job:
 https://url.corp.redhat.com/2413e8e

**Comment by petr-balogh** [comment]
New verification job:
https://url.corp.redhat.com/77ea28d

File: ocs_ci/utility/vsphere.py [hunk 1]
```diff
@@ -785,8 +785,19 @@ def check_folder_exists(self, name, cluster, dc):
...
+                    _rc = True
+            except vmodl.fault.ManagedObjectNotFound as ex:
+                logger.exception(
+                    "There was an exception hit while attempting to check if folder exists!"
+                )
+                if (
+                    "has already been deleted or has not been completely create"
+                    not in str(ex)
+                ):
+                    raise
```
Comment (shivamdurgbuns) [inline ocs_ci/utility/vsphere.py:799]: it would be better if we describe which exception to raise here

**Review by dahorak — APPROVED** [review]

File: ocs_ci/utility/vsphere.py (hunk 1 above)
Comment (petr-balogh) [reply ocs_ci/utility/vsphere.py:799]: Whole exception details will be described as part of looger.exception as it automatically include exception details.
```
logger.exception(
```

The `raise` is just simple re-raise of exception caught in except block, so nothing to specify here IMO.

**Review by shylesh — APPROVED** [review]
LGTM

**Pull request merged**